pkg/vault/contracts/VaultFactory.sol

```diff
@@ -64,28 +74,67 @@ contract VaultFactory is Authentication {
      * @param targetAddress Expected Vault address. The function will revert if the given salt does not deploy the
      * Vault to the target address.
      */
-    function create(bytes32 salt, address targetAddress) external authenticate {
-        if (isDisabled) {
-            revert VaultAlreadyCreated();
+    function create(
+        bytes32 salt,
+        address targetAddress,
+        bytes memory vaultCreationCode,
+        bytes memory vaultAdminCreationCode,
+        bytes memory vaultExtensionCreationCode
+    ) external authenticate {
+        if (vaultCreationCodeHash != keccak256(vaultCreationCode)) {
+            revert InvalidBytecode("Vault");
+        } else if (vaultAdminCreationCodeHash != keccak256(vaultAdminCreationCode)) {
+            revert InvalidBytecode("VaultAdmin");
+        } else if (vaultExtensionCreationCodeHash != keccak256(vaultExtensionCreationCode)) {
+            revert InvalidBytecode("VaultExtension");
         }
-        isDisabled = true;
 
         address vaultAddress = getDeploymentAddress(salt);
         if (targetAddress != vaultAddress) {
             revert VaultAddressMismatch();
         }
 
-        VaultAdmin vaultAdmin = new VaultAdmin(
-            IVault(vaultAddress),
-            _pauseWindowDuration,
-            _bufferPeriodDuration,
-            _minTradeAmount,
-            _minWrapAmount
+        VaultAdmin vaultAdmin = VaultAdmin(
+            payable(
+                Create2.deploy(
```

Comment:
It'd be nice to have a comment on why we're using payable and `create2` here.

Reply:
I guess because we don't care what the address is for the proxy contracts, only the main Vault?

Reply:
yes, we don't care about addresses for the proxy contracts